doc/nvim-tree-lua.txt

```diff
@@ -324,6 +328,11 @@ Subsequent calls to setup will replace the previous configuration.
         show_on_open_dirs = true,
         timeout = 400,
       },
+      modified = {
+        enable = false,
+        show_on_dirs = true,
+        show_on_open_dirs = false,
```

Comment:
I actually wanted to change git & diagnostics to the same as modified because I think that'd be the default that most people want, but didn't cause i thought that'd be breaking.

Idk what is this plugin's stance on changes in default though, if you think it's alright I can make that change in this PR.

In the mean time I'll change modified to be consistent with git for now :).

Reply:
Yeah, I'd like to change those defaults, they make more sense. That's what I use.

Users do get upset when we make changes like that...

I am open to trying it again: how about we make a separate PR for those sensible defaults that we can revert if there is upset.

Reply:
I have this idea in my head for a while now. Wanted to open an issue but didn't find time to do investigation deep enough to justify it. 
I would like to discuss future of the project and breaking changes moving forward. I wanted to argue in favour of releasing pre-release version with multiple breaking changes: removed/unified functionality, reorganised configuration options, general clean up. I think project could greatly benefit from refactor without worrying about breaking changes. It would then be released as version 2 for example. Would there be any appetite for that at all?

Reply:
> reorganised configuration options, general clean up

Yes. `git.ignore` not being in filters. `renderer` too large. `diagnostics.icons` not in `icons.glyphs`. Some root options not in `view` or their own section.

> I would like to discuss future of the project and breaking changes moving forward.

We can break things in a controlled manner that will result in a better experience. We are also good at refactoring options, although we don't use it much.

> I wanted to argue in favour of releasing pre-release version with multiple breaking changes

> It would then be released as version 2 for example.

Many lua plugins and packer are getting very good at handling versions. I think it is time.

`master` could remain as "version 1" to avoid interruptions to users, however we could add `devel`, `weekly` etc. along with `stable`, `stable-2`, `stable-2.1`, `stable-2.1.3` etc. I like SEMVER.

Reply:
I did discover an interesting pattern with lualine:

User is prompted to run `:LualineNotices` when there is a configuration problem.

```
theme(base16): nvim-base16 is not currently present in your runtimepath, make sure it is properly installed, fallback to default colors.
```

Reply:
That's great to hear. Let's open a dedicated issue to discuss it further.

Reply:
https://github.com/nvim-tree/nvim-tree-management